src/libslic3r/GCode/ToolOrdering.cpp

```diff
@@ -757,24 +757,17 @@ void ToolOrdering::reorder_extruders_for_minimum_flush_volume()
     const unsigned int number_of_extruders = (unsigned int) (sqrt(flush_matrix.size()) + EPSILON);
     // Extract purging volumes for each extruder pair:
     std::vector<std::vector<float>> wipe_volumes;
-    if (m_print_config_ptr->purge_in_prime_tower) {
```

Comment:
Nice catch.
The condition here should be  `m_print_config_ptr->purge_in_prime_tower || is_bbl_machine`

With that said, this shouldn't cause crash as worst case here is the tool ordering is not optimal because the wipe_volume is correct.
Any idea what why this caused the crash?

Reply:
The m_print_config_ptr was a null pointer - it should be replaced with print_config which is set at the start of the function now :) I'll partly revert the change and incorporate the is_BBL check. 

It makes a big difference optimising the tool ordering at least for BBL printers - take a look at the screenshots below ;) Both in material and time saved (hence why I also increased the max extruder size to 12 from 8).

Reply:
So the is_bbl variable/function is not present in the tool ordering class or any of its immediate access to, unless Im missing something? (i.e. it doesnt exist in the print config variable). Any ideas?

Personally I'm inclined to use the submitted code as I can't imagine a scenario where the tool ordering shouldn't be done based on the flush matrix volumes (as these define how much waste there is going to be).

Reply:
> The m_print_config_ptr was a null pointer - it should be replaced with print_config which is set at the start of the function now :) I'll partly revert the change and incorporate the is_BBL check.

That's strange, I can't repro it with your attached project on latest dev build

> It makes a big difference optimising the tool ordering at least for BBL printers - take a look at the screenshots below ;) Both in material and time saved (hence why I also increased the max extruder size to 12 from 8).

Okay, sounds good

Reply:
> > The m_print_config_ptr was a null pointer - it should be replaced with print_config which is set at the start of the function now :) I'll partly revert the change and incorporate the is_BBL check.
> 
> That's strange, I can't repro it with your attached project on latest dev build
> 
> > It makes a big difference optimising the tool ordering at least for BBL printers - take a look at the screenshots below ;) Both in material and time saved (hence why I also increased the max extruder size to 12 from 8).
> 
> Okay, sounds good
Use this project here: 
[crash.3mf.zip](https://github.com/SoftFever/OrcaSlicer/files/13743965/crash.3mf.zip)

The one below was just for the purge volumes :)

Reply:
> Personally I'm inclined to use the submitted code as I can't imagine a scenario where the tool ordering shouldn't be done based on the flush matrix volumes (as these define how much waste there is going to be and we want to minimise that).

when `purge_in_prime_tower` is not specified, wipe tower won't serve as purge tower hence the matix is not used(for non BBL)
That's that rational behind the logic here.

> So the is_bbl variable/function is not present in the tool ordering class or any of its immediate access to, unless Im missing something? (i.e. it doesnt exist in the print config variable). Any ideas?

We can query and store it as a member in constructor through Print object

Reply:
gotcha. let me have a go at it.

Reply:
Done